src/utils/use-refresh-on-focus.ts

```diff
@@ -0,0 +1,23 @@
+import {useIsFocused} from '@react-navigation/native';
+import {differenceInSeconds} from 'date-fns';
+import {useEffect} from 'react';
+
+export function useRefreshOnFocus(
+  tick: Date | undefined,
+  hardRefreshLimitInSeconds: number,
+  hardRefresh: () => void,
+  softRefreshLimitInSeconds: number,
+  softRefresh: () => void,
+) {
+  const isFocused = useIsFocused();
+  useEffect(() => {
+    if (!isFocused || !tick) return;
+    const timeSinceLastTick = differenceInSeconds(Date.now(), tick);
+
+    if (timeSinceLastTick >= hardRefreshLimitInSeconds) {
+      hardRefresh();
+    } else if (timeSinceLastTick >= softRefreshLimitInSeconds) {
+      softRefresh();
+    }
+  }, [isFocused]);
```

Comment:
Seems kinda strange to have so many parameters used in the `useEffect` which is not in the dependency list. Might be some real reason for it, but just commenting it here, so you can take an extra look if it is necessary to exclude them.

Reply:
Generally the rule is that they should be provided:
https://beta.reactjs.org/reference/react/useEffect#specifying-reactive-dependencies

React provides a lint rule "exhaustive-deps" to check for this, but we haven't enabled it since we break it so many places.

Reply:
👍 Updated now